src/rpdk/core/project.py

```diff
@@ -432,10 +447,20 @@ def _set_docs_properties(self, propname, prop, proppath):
             template = self.env.get_template("docs-subproperty.md")
             docs_path = self.root / "docs"
 
-            prop["properties"] = {
-                name: self._set_docs_properties(name, value, proppath + (name,))
-                for name, value in prop["properties"].items()
-            }
+            if "properties" in prop:
+                prop["properties"] = {
+                    name: self._set_docs_properties(name, value, proppath + (name,))
+                    for name, value in prop["properties"].items()
```

Comment:
```
try:
    prop["properties"] = {
                    name: self._set_docs_properties(name, value, proppath + (name,))
                    for name, value in prop.get("properties", prop["patternProperties"])
                }
except KeyError:
    raise InvalidSchemaPropertyError(
                    "Object property require a properties or patternProperties field"
                )
```
 could consolidate this logic nicely without all the branching.  i don't know if we need to even worry about a schema where neither properties or patternproperties is specified, but it guess it doesn't hurt including it

Reply:
The problem with the solution is that `prop["patternProperties"]` will be evaluated before dict fallback happens so the KeyError always happens.

This is what would be in JS
```js
prop.get("properties", () => prop["patternProperties"])
```

Possibile solutions:
```python
            try:
                object_properties = prop["properties"]
                  if "properties" in prop else prop["patternProperties"]
            except KeyError:
                raise InvalidSchemaPropertyError(
                    "Object property requires a properties or patternProperties field"
                )

            prop["properties"] = {
                name: self._set_docs_properties(name, value, proppath + (name,))
                for name, value in object_properties.items()
            }
```

```python
            try:
                prop["properties"] = {
                    name: self._set_docs_properties(name, value, proppath + (name,))
                    for name, value in prop.get(
                        "properties", prop.get("patternProperties")
                    ).items()
                }
            except AttributeError:
                raise InvalidSchemaPropertyError(
                    "Object property requires a properties or patternProperties field"
                )

```

or the above with default to empty list

Reply:
I pushed the first possible solution, let me know what you think :)